src/diffusion/model.py

```diff
@@ -11,25 +11,22 @@
 
 def get_components(name: str, mode: str, **model_kwargs):
     model = AutoModel.from_pretrained(name)
-    emb_dim = model.config.d_model
 
     if mode == "same":
-        return model.encoder, model.decoder, emb_dim
+        return model.encoder, model.decoder, model.config.d_model
 
     elif mode == "bert":
         decoder_config = BertConfig(
             vocab_size=model.config.vocab_size,
             is_decoder=True,
-            hidden_size=emb_dim,
-            num_attention_heads=model.config.num_attention_heads,
             add_cross_attention=True,
             **model_kwargs,
         )
         zero_rank_info(f"BERT config:\n{decoder_config}")
 
         decoder = AutoModel.from_config(decoder_config)
 
-        return model.encoder, decoder, emb_dim
+        return model.encoder, decoder, model_kwargs["hidden_size"]
```

Comment:
Let's use `decoder_config.hidden_size` instead of `model_kwargs["hidden_size"]`. In case of default hyperparameters, model kwargs may not contain this value